server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -190,4 +190,6 @@ public final String simpleName() {
      *  Both {@code this} and {@code mergeWith} will be left unmodified. */
     public abstract Mapper merge(Mapper mergeWith);
 
+    public abstract void validate(DocumentFieldMappers mappers);
```

Comment:
Probably worth javadoc just because all the mappers are going to have to deal with it.

Reply:
++